### ISSUE TITLE
Unable to find modules (source_modules and node_modules are `None`)

### ISSUE THREAD

**Issue by mijelin** [unlight/sublime-import-helper#45]
# Issue
Getting the `No imports found for *` message when I try to auto-import.

I've tried uninstalling and reinstalling (it had previously been working fine).

# Debug info
## Sublime console output

Per https://github.com/unlight/sublime-import-helper/issues/32#issuecomment-303237359, I enabled debugging in `utils.py` and ran `Import Helper: Update imports`:
```
# Note: <SOURCE_FOLDER> is the root of my project + contains package.json

source_folders: ['<SOURCE FOLDER>']
import_root: <SOURCE FOLDER>
Run command: ['get_folders', {'folders': ['<SOURCE FOLDER>']}]
Run command: ['get_modules', {'importRoot': '<SOURCE FOLDER>', 'packageKeys': ['devDependencies']}]
Run command: ['get_modules', {'importRoot': '<SOURCE FOLDER>', 'packageKeys': ['dependencies']}]
Trying to decode: 
Exception in thread Thread-47:
Traceback (most recent call last):
  File "./python3.3/threading.py", line 901, in _bootstrap_inner
  File "./python3.3/threading.py", line/858, in run
  File "~/Library/Application Support/Sublime Text 3/Packages/ImportHelper/utils.py", line 40, in run_command
    return callback(None, result)
  File "~/Library/Application Support/Sublime Text 3/Packages/ImportHelper/import_helper.py", line 80, in get_modules_callback
    node_modules.extend(result)
TypeError: 'NoneType' object is not iterable

Trying to decode: 
Exception in thread Thread-46:
Traceback (most recent call last):
  File "./python3.3/threading.py", line 901, in _bootstrap_inner
  File "./python3.3/threading.py", line 858, in run
  File "~/Library/Application Support/Sublime Text 3/Packages/ImportHelper/utils.py", line 40, in run_command
    return callback(None, result)
  File "~/Library/Application Support/Sublime Text 3/Packages/ImportHelper/import_helper.py", line 80, in get_modules_callback
    node_modules.extend(result)
TypeError: 'NoneType' object is not iterable

Trying to decode: 
Exception in thread Thread-45:
Traceback (most recent call last):
  File "./python3.3/threading.py", line 901, in _bootstrap_inner
  File "./python3.3/threading.py", line 858, in run
  File "~/Library/Application Support/Sublime Text 3/Packages/ImportHelper/utils.py", line 40, in run_command
    return callback(None, result)
  File "~/Library/Application Support/Sublime Text 3/Packages/ImportHelper/import_helper.py", line 31, in get_source_modules_callback
    for item in modules:
TypeError: 'NoneType' object is not iterable
```

## Sublime project settings
```
{
  "folders":
  [
    {
      "path": "."
    }
  ]
} 
```

## ImportHelper plugin settings
All default --
```
{
	"insert_position": "end",
	"from_quote": "'",
	"space_around_braces": true
}
```

## Versions
Mac OS 10.11.6
Sublime Text 3 Build 3143

```
~% which node
/usr/local/bin/node

~% node -v
v7.7.2
```

**Comment by unlight** [comment]
Looks like `esm-exports` some how returns Promise.reject(undefined).
In this case callback(err, result) will be executed.
`err` will be undefined and `result` undefined, undefined will be converted to python None type.

Previous version of `esm-exports` swallowed all errors while parsing and returned empty array.
It's bad pattern in general, need to catch this errors some how.
I need to think how we can handle unknown errors more gracefully.

**Comment by unlight** [comment]
Please, could you give a list of `devDependencies` and `dependencies` of your package.json?

**Comment by mijelin** [comment]
Sure!

```
  "dependencies": {
    "@agm/core": "^1.0.0-beta.2",
    "@angular/animations": "^4.4.4",
    "@angular/common": "^4.4.4",
    "@angular/compiler": "^4.4.4",
    "@angular/compiler-cli": "^4.4.4",
    "@angular/core": "^4.4.4",
    "@angular/forms": "^4.4.4",
    "@angular/http": "^4.4.4",
    "@angular/platform-browser": "^4.4.4",
    "@angular/platform-browser-dynamic": "^4.4.4",
    "@angular/router": "^4.4.4",
    "@ng-bootstrap/ng-bootstrap": "^1.0.0-beta.4",
    "airbrake-js": "^0.9.8",
    "angular-svg-round-progressbar": "https://registry.npmjs.org/angular-svg-round-progressbar/-/angular-svg-round-progressbar-1.1.0.tgz",
    "autolinker": "http://registry.npmjs.org/autolinker/-/autolinker-1.6.0.tgz",
    "bootstrap": "^4.0.0-beta",
    "classlist.js": "^1.1.20150312",
    "core-js": "^2.4.1",
    "d3-axis": "^1.0.7",
    "d3-scale": "^1.0.6",
    "d3-selection": "^1.1.0",
    "d3-time-format": "http://registry.npmjs.org/d3-time-format/-/d3-time-format-2.0.5.tgz",
    "d3-transition": "^1.1.0",
    "lory.js": "^2.2.1",
    "rxjs": "^5.4.0",
    "time-ago-pipe": "^1.2.1",
    "web-animations-js": "^2.3.1",
    "zone.js": "^0.8.18"
  },
  "devDependencies": {
    "@angular/cli": "^1.5.0",
    "@types/d3-axis": "^1.0.8",
    "@types/d3-scale": "^1.0.9",
    "@types/d3-selection": "^1.1.0",
    "@types/d3-time-format": "^2.0.5",
    "@types/d3-transition": "^1.1.0",
    "@types/jasmine": "^2.5.53",
    "@types/lory.js": "https://registry.npmjs.org/@types/lory.js/-/lory.js-0.4.28.tgz",
    "@types/node": "^6.0.45",
    "codelyzer": "^3.2.0",
    "cors": "^2.7.1",
    "del": "^2.2.0",
    "husky": "^0.13.4",
    "intl": "1.2.5",
    "istanbul-instrumenter-loader": "^2.0.0",
    "jasmine": "^2.6.0",
    "jasmine-core": "^2.6.4",
    "jasmine-spec-reporter": "^4.1.0",
    "jquery": "3.0.0",
    "karma": "^1.7.1",
    "karma-chrome-launcher": "^2.1.1",
    "karma-cli": "^1.0.1",
    "karma-coverage-istanbul-reporter": "^1.0.0",
    "karma-firefox-launcher": "^1.0.0",
    "karma-jasmine": "^1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "karma-phantomjs-launcher": "^1.0.4",
    "karma-sourcemap-loader": "^0.3.4",
    "lodash": "^4.11.1",
    "phantomjs-prebuilt": "^2.1.15",
    "popper.js": "^1.11.0",
    "protractor": "^5.1.1",
    "supports-color": "^3.1.2",
    "tab-navigation-fix": "^1.0.0",
    "ts-node": "^3.0.2",
    "tslint": "^5.8.0",
    "typescript": "2.3.4"
  },
```

**Comment by mijelin** [comment]
One of the recent releases appears to have fixed this (working fine on 1.7.6, possibly earlier).

Thanks!
